create-function-deps_save_and_drop_dependencies.sql

```diff
@@ -0,0 +1,212 @@
+    WHERE
+        schemaname = v_curr.obj_schema
+        AND matviewname = v_curr.obj_name;
+    
+    --save index/unique index: 
+    INSERT INTO public.deps_saved_ddl(deps_view_schema, deps_view_name, deps_ddl_to_run)
+    SELECT
+        p_view_schema,
+        p_view_name,
+        indexdef || ';' AS deps_ddl_to_run
+    FROM pg_indexes
+    WHERE
+        schemaname = v_curr.obj_schema
+        AND tablename = v_curr.obj_name;
+
+    --save mat view owner: 
+    INSERT INTO public.deps_saved_ddl(deps_view_schema, deps_view_name, deps_ddl_to_run)
+    SELECT
+        p_view_schema,
+        p_view_name,
+        'ALTER MATERIALIZED VIEW ' || v_curr.obj_schema || '.' || v_curr.obj_name
+        || ' OWNER TO ' || matviewowner || ';' AS deps_ddl_to_run
+    FROM pg_matviews
+    WHERE
+        schemaname = v_curr.obj_schema
+        AND matviewname = v_curr.obj_name;
+
+END IF;
+
+--save comments on dependencies
+INSERT INTO public.deps_saved_ddl(deps_view_schema, deps_view_name, deps_ddl_to_run)
+SELECT
+    p_view_schema,
+    p_view_name,
+    'COMMENT ON ' ||
+    CASE
+        WHEN c.relkind = 'v' THEN 'VIEW'
+        WHEN c.relkind = 'm' THEN 'MATERIALIZED VIEW'
+        else ''
+    END || ' ' || n.nspname || '.' || c.relname || ' IS ''' || replace(d.description, '''', '''''') || ''';'
+FROM pg_class c
+JOIN pg_namespace n ON n.oid = c.relnamespace
+JOIN pg_description d on
+    d.objoid = c.oid
+    AND d.objsubid = 0
+WHERE
+    n.nspname = v_curr.obj_schema
+    AND c.relname = v_curr.obj_name
+    AND d.description is not null;
+
+--save comments on dependency columns
+INSERT INTO public.deps_saved_ddl(deps_view_schema, deps_view_name, deps_ddl_to_run)
+SELECT
+    p_view_schema,
+    p_view_name,
+    'COMMENT ON COLUMN ' || n.nspname || '.' || c.relname || '.' || a.attname
+    || ' IS ''' || replace(d.description, '''', '''''') || ''';' AS deps_ddl_to_run
+FROM pg_class AS c
+JOIN pg_attribute AS a ON c.oid = a.attrelid
+JOIN pg_namespace AS n ON n.oid = c.relnamespace
+JOIN pg_description AS d ON
+    d.objoid = c.oid
+    AND d.objsubid = a.attnum
+WHERE
+    n.nspname = v_curr.obj_schema
+    AND c.relname = v_curr.obj_name
+    AND d.description IS NOT NULL;
+
+--save permissions on object
+INSERT INTO public.deps_saved_ddl(deps_view_schema, deps_view_name, deps_ddl_to_run)
+SELECT
+    p_view_schema,
+    p_view_name,
+    'GRANT ' || privilege_type || ' ON ' || table_schema || '.'
+    || table_name || ' TO ' || grantee || ';' AS deps_ddl_to_run
+FROM information_schema.role_table_grants
+WHERE
+    table_schema = v_curr.obj_schema
+    AND table_name = v_curr.obj_name;
+
+IF dryrun IS FALSE THEN
+    EXECUTE 'DROP ' ||
+    CASE
+        WHEN v_curr.obj_type = 'v' THEN 'VIEW'
+        WHEN v_curr.obj_type = 'm' THEN 'MATERIALIZED VIEW'
+    END || ' ' || v_curr.obj_schema || '.' || v_curr.obj_name || ';';
+END IF;
+
+END loop;
+
+END;
+$$;
+
+ALTER FUNCTION public.deps_save_and_drop_dependencies OWNER TO natalie;
```

Comment:
Can you add (VARCHAR, VARCHAR, BOOLEAN) to the end of the function as well? To make it explicit